### PR TITLE
INT-572 Have server-side integrations handle ajs unbundled messages

### PR DESCRIPTION
**Pull request by tsholmes**

### CODE REVIEW

File: lib/proto.js [hunk 1]
```diff
@@ -108,8 +108,22 @@ exports.initialize = function(){
...
+  if (facade.channel() === 'client') {
+    // If the bundled metadata isn't included in the message, the client sending
+    // this message didn't support unbundling, so it must have already been sent
+    // from the client integration.
+    var bundled = facade.proxy('_metadata.bundled');
+    if (bundled && !bundled[this.name]) {
+      return true;
+    }
+  }
+  return this.channels.indexOf(facade.channel()) !== -1;
```
Comment (ndhoule) [inline lib/proto.js]: Slightly more expressive:

https://github.com/ndhoule/includes

It's a native method in later node versions but not in the version integrations are using

File: lib/proto.js [hunk 2]
```diff
@@ -108,8 +108,22 @@ exports.initialize = function(){
...
+  // - this event came from the browser
+  // - this integration wasn't bundled in analytics.js
+  if (facade.channel() === 'client') {
+    // If the bundled metadata isn't included in the message, the client sending
+    // this message didn't support unbundling, so it must have already been sent
+    // from the client integration.
+    var bundled = facade.proxy('_metadata.bundled');
+    if (bundled && !bundled[this.name]) {
+      return true;
+    }
```
Comment (ndhoule) [inline lib/proto.js]: I think before merging this, we should sit down and talk a little bit about replay scenarios more to make sure we're covering all use cases

File: test/proto.js [hunk 3]
```diff
@@ -735,6 +735,36 @@ describe('proto', function(){
...
     });
   });
+
+  describe('.enabled()', function(){
+    var Test = integration('test').server();
+    var test = Test({});
+
+    it('should return true when sent on a supported channel', function(){
+      var facade = new Page({ channel: 'server' });
+      assert(true === test.enabled(facade));
```
Comment (ndhoule) [inline test/proto.js]: In general, prefer `assert.strictEquals(x, y)` over `assert(x === y)`. It gives you way more information when the test fails

**Comment by tsholmes** [comment]
Based on changes in https://github.com/segment-integrations/analytics.js-integration-segmentio/pull/17 and https://github.com/segmentio/cdn/pull/105

File: test/proto.js [hunk 4]
```diff
@@ -735,6 +735,45 @@ describe('proto', function(){
       assert(false == segment.retry(new TypeError('whoops')));
     });
   });
+
+  describe.only('.enabled()', function(){
```
Comment (ndhoule) [inline test/proto.js]: Remove `.only`

**Pull request merged**